fs-index/src/index.rs

```diff
@@ -515,6 +531,7 @@ impl<Id: ResourceId> ResourceIndex<Id> {
                 self.id_to_paths.remove(&id.item);
             }
 
+            result.removed.insert(id.item);
```

Comment:
So, if I'm not mistaken, this line should be moved under the condition above:
```
if self.id_to_paths[&id.item].is_empty() {
    self.id_to_paths.remove(&id.item);
    // emit the removed event
}
```

I.e. only removal of the last path with same id should be considered real removal.

And we could log duplicates removal in `else` branch of same condition.